core/src/main/scala/com/databricks/labs/remorph/discovery/TSqlTableDefinitions.scala

```diff
@@ -0,0 +1,214 @@
+package com.databricks.labs.remorph.discovery
+
+import com.databricks.labs.remorph.intermediate.{ColumnDetail, DataType}
+import com.databricks.labs.remorph.parsers.tsql.{DataTypeBuilder, TSqlLexer, TSqlParser}
+import org.antlr.v4.runtime.{CharStreams, CommonTokenStream}
+
+import java.sql.Connection
+import scala.collection.mutable
+
+class TSqlTableDefinitions(conn: Connection) {
+
+  /**
+   * Parses a data type string and returns the corresponding DataType object.
+   *
+   * @param dataTypeString The string representation of the data type.
+   * @return The DataType object corresponding to the input string.
+   */
+  private def getDataType(dataTypeString: String): DataType = {
+    val inputString = CharStreams.fromString(dataTypeString)
+    val lexer = new TSqlLexer(inputString)
+    val tokenStream = new CommonTokenStream(lexer)
+    val parser = new TSqlParser(tokenStream)
+    val ctx = parser.dataType()
+    val dataTypeBuilder = new DataTypeBuilder
+    dataTypeBuilder.build(ctx)
+  }
+
+  private def getTableDefinitionQuery(catalogName: String): String = {
+    s"""WITH column_info AS (
+       |  SELECT
+       |      TABLE_CATALOG,
+       |      TABLE_SCHEMA,
+       |      TABLE_NAME,
+       |      STRING_AGG(
+       |              CONCAT(
+       |                      column_name, '§',
+       |                      CASE
+       |                          WHEN numeric_precision IS NOT NULL AND numeric_scale IS NOT NULL
```

Comment:
use two-space indentation **everywhere**, including the embedded query literals. here you use two-space for the first line, but all other lines use four-space. **this is not consistent and not professional.**

Reply:
done.